### DISCUSSION TITLE
This library can't work on ESP32-S3-Box?

### DISCUSSION THREAD

**Discussion by yiming564** [schreibfaul1/ESP32-audioI2S#532] [Q&A] [answered]
I made my own ESP32-S3-WROOM-1U-N16R8 dev board and I wanted to use this library on PlatformIO.
In order to use PSRAM, I set the board as esp32s3box in platformio.ini:
`

; PlatformIO Project Configuration File
;
;   Build options: build flags, source filter
;   Upload options: custom upload port, speed and extra flags
;   Library options: dependencies, extra library storages
;   Advanced options: extra scripting
;
; Please visit documentation for the other options and examples
; https://docs.platformio.org/page/projectconf.html

[env:esp32s3box]
platform = espressif32
board = esp32s3box
framework = arduino
upload_speed = 921600
monitor_speed = 115200
build_flags = -D BOARD_HAS_PSRAM -O3
board_build.partitions = partitions.csv
lib_deps = 
	bodmer/TFT_eSPI@^2.5.23
	https://github.com/schreibfaul1/ESP32-audioI2S
	paulstoffregen/XPT2046_Touchscreen@0.0.0-alpha+sha.26b691b2c8
	adafruit/RTClib@^2.1.1
	lvgl/lvgl@^8.3.4

`
When I set the board as esp32-s3-devkitc-1, this library worked well. However, the program will crash if I change the board type.
This is really strange...

This is a part of my schematic:
![image](https://github.com/schreibfaul1/ESP32-audioI2S/assets/78789251/320b1df0-5c80-43ff-b781-045c59ba7e1e)
![image](https://github.com/schreibfaul1/ESP32-audioI2S/assets/78789251/78dd03d1-c39d-488d-80e3-942292d8ec21)

**Comment by yiming564** [comment]
By the way, the crash is because of TG1WDT_SYS_RST

**Comment by yiming564** [comment]
uint32_t Audio::stopSong() {
	Serial0.printf("%s:%d\n", __FILE__, __LINE__);
    uint32_t pos = 0;
    if(m_f_running) {
        m_f_running = false;
        if(getDatamode() == AUDIO_LOCALFILE){
            m_streamType = ST_NONE;
            pos = getFilePos() - inBufferFilled();
            audiofile.close();
            AUDIO_INFO("Closing audio file");
        }
    }
    if(audiofile){
        // added this before putting 'm_f_localfile = false' in stopSong(); shoulf never occur....
        audiofile.close();
        AUDIO_INFO("Closing audio file");
        log_w("Closing audio file");  // for debug
    }
	Serial0.printf("%s:%d\n", __FILE__, __LINE__);
    memset(m_outBuff, 0, 2048 * 2 *sizeof(uint16_t));     // The program crashed here
	Serial0.printf("%s:%d\n", __FILE__, __LINE__);
    i2s_zero_dma_buffer((i2s_port_t) m_i2s_num);
    return pos;
}

**Reply by schreibfaul1** [reply]
m_outBuff is a pointer to a memory area allocated in the constructor. A crash would mean that the value of m_outBuff was changed?
TG1WDT_SYS_RST is more of a reset due to unserviced watchdog.

**Reply by yiming564** [reply]
I found that the program didn't call the constructor at all...Maybe it is because I set Audio as a global variable, or maybe it is the bug of the compiler.
BUT, after manually called the constructor, the program still crashed at memset(m_outBuff, 0, 2048 * 2 *sizeof(uint16_t)); I am sure that m_outBuff wasn't wild pointer or null pointer.

**Reply by yiming564** [reply, marked as answer]
![image](https://github.com/schreibfaul1/ESP32-audioI2S/assets/78789251/d320c0f4-6ec4-4574-ac01-7ba26648ab3d)
emmm....
I used the wrong pins and that is the problem, I need to fix my PCB now...
